R/simPWSurv.R

```diff
@@ -117,14 +117,14 @@ simPWSurv <- function(n=100,
          group_by(Stratum) %>% mutate(Treatment=fixedBlockRand(n=n(),block=block))  %>% # assign treatment
          # generate time to failure and time to dropout
          dplyr::group_by(Stratum,Treatment)
-    utr <- unique(x$Treatment)
     usr <- unique(x$Stratum)
+    utr <- unique(x$Treatment)
     x$failTime <- 0
     x$dropoutTime <- 0
     for(sr in usr){for(tr in utr){
       indx <- x$Stratum==sr & x$Treatment==tr
-      x$failTime[indx] <- rpwexp(n=sum(indx),failRates=filter(failRates,Stratum==sr&Treatment==tr))
-      x$dropoutTime[indx] <- rpwexp(n=sum(indx),failRates=filter(dropoutRates,Stratum==sr&Treatment==tr))
+      x$failTime[indx] <- rpwexpinvRcpp(n=sum(indx),failRates=failRates[failRates$Stratum==sr&failRates$Treatment==tr,])
```

Comment:
You probably want to use `drop = FALSE` in all places where the `[` subsetting operator is used: https://adv-r.hadley.nz/subsetting.html#simplify-preserve